modules/core/src/main/scala/fs2/kafka/consumer/KafkaConsume.scala

```diff
@@ -66,4 +66,20 @@ trait KafkaConsume[F[_], K, V] {
     */
   def partitionsMapStream
     : Stream[F, Map[TopicPartition, Stream[F, CommittableConsumerRecord[F, K, V]]]]
+
+  /**
+    * Stops consuming new messages from Kafka.
+    * This method could be used to implement a graceful shutdown.<br>
+    * <br>
+    * This method has a few effects:
+    * 1. After this call no more data will be fetched from Kafka through the `poll` method.
+    * 2. All currently running streams will continue to run until all in-flight messages will be processed.
+    *    It means that streams will be completed when all fetched messages will be processed.<br>
+    * <br>
+    * If some of the [[stream]] methods will be called after [[stopConsuming]] call,
+    * these methods will return empty streams.<br>
+    * <br>
+    * More than one call of [[stopConsuming]] will have no effect.
+    */
+  def stopConsuming: F[Unit]
```

Comment:
I think I'd tend to put this in the `Lifecycle` trait rather than the `Consume` trait, but not sure - any thoughts? (cc @vlovgr)

I'm happy to make and push that change if you (@LMnet) are happy with that, since it's my fault this PR needed updating 🙂

Reply:
I hesitated between `KafkaConsume` and `KafkaLifecycle` and decided that `KafkaConsume` is better for this method because it's highly coupled with the stream methods.

Reply:
I'm also leaning slightly towards `KafkaConsume`, but I'm ultimately happy either way. :)